src/binary-reader.c

```diff
@@ -1694,12 +1705,15 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
     uint32_t i, num_relocs, section;
     in_u32_leb128(ctx, &section, "section");
     in_u32_leb128(ctx, &num_relocs, "relocation count count");
-    CALLBACK(on_reloc_count, num_relocs);
+    WASM_ZERO_MEMORY(section_name);
+    if (section == WASM_BINARY_SECTION_CUSTOM)
+      in_str(ctx, &section_name, "section name");
+    CALLBACK(on_reloc_count, num_relocs, section, section_name);
```

Comment:
Agreed.  It should be other away around as the "spec" says: https://github.com/WebAssembly/tool-conventions/blob/master/Linking.md.

I will fix